grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptData.java

```diff
@@ -176,7 +176,9 @@ private void generateRuleLHS(Representation halResource, Rule rule) {
 
     private void generateOwnerInstances(Representation halResource, Resource<?> conceptResource, int separationDegree) {
         final TypeName roleType = conceptResource.type().getName();
-        conceptResource.ownerInstances().forEach(instance -> {
+        Stream<Instance> ownersStream = conceptResource.ownerInstances().stream().skip(offset);
```

Comment:
So here is something that sucks. `ownerInstances` serialises everything into a set which means we still reading all the instances from disk despite skipping over a few. If we wanted to make this faster we would need to get `ownerInstances` to lazy load and if we doing that I cote for making it return a `Stream`